CODE-OF-CONDUCT.md

```diff
@@ -14,6 +16,40 @@ We follow the [CNCF Contributor Code of Conduct](https://github.com/cncf/foundat
+Much like subtle-isms, nearly everyone will on occassion dip into Yellow. To err is human. It shows emotional maturity to recognize when this has happened and apologize for the behavior. It is a community concern if the situation keeps repeating, resulting in a warning and eventual banning.
+
+Red:
+Either frequent repeating of Yellow, or outright harassment, insults, etc. as listed non-exhaustively in Principles.
+These interactions contribute to a toxic environment, are disruptive to productivity, and cause an unwelcome and unsafe atmosphere.
+Red interactions are likely to result in at least temporary bans.
+
+Threats and violence will result in immediate permanent bans and involvement of law enforcement.
+
 ### Moderation
 
-- If you feel any of SPIFFE's communication channels require moderation, please e-mail the [SPIFFE Steering Committee (SSC)](mailto:ssc@spiffe.io). The SSC will issue a warning to users who don't follow this code of conduct. A second offense results in a temporary ban. A third offense warrants a permanent ban. It is at the moderator's discretion to un-ban offending users, or to immediately ban a toxic user without warning.
+If you feel any of SPIFFE's communication channels require moderation, or any interaction you witness with other community members online or in-person requires reporting, please e-mail the [SPIFFE Steering Committee (SSC)](mailto:ssc@spiffe.io).
+
+The SSC will investigate all reports in a reasonable timeframe.
+The SSC encourages timely reporting so that any potential evidence or other investigations will be more viable.
+At the end of an investigation, the SSC will respond to reporters with the high-level conclusion of any action resulting from their report. If you would like clarity on a decision, the SSC is happy to discuss. If you are unsatisfied and still believe the situation requires escalation, report to [the CNCF Code of Conduct Committee](https://www.cncf.io/conduct/committee/).
+
+Retaliation against Code of Conduct reporters will result in a permanent ban.
+
+The SSC confidentially tracks all reports centrally, discussing information only with other CoC bodies under CNCF/Linux Foundation and only on an as-needed basis. All past SSC members are expected to continue to respect this confidentiality indefinitely- the SSC will consider it an egregious Code of Conduct violation to expose such information.
+
+When discussing violations with a user, the SSC will make reasonable attempts to generalize information so that it is still useful but not identifying of the reporter. This will not always be possible (eg., if a report is made regarding a 1-on-1 interaction).
+
+The SSC will issue a warning to users who don't follow this code of conduct. A second offense results in a temporary ban. A third offense warrants a permanent ban. It is at the SSC's discretion to un-ban offending users, or to immediately ban a toxic user without warning.
```

Comment:
I get the desire to be able to unilaterally ban a toxic user without warning... but that may go a little too far except in the most extreme case (maybe just the violence use case listed above). We may not want to reserve the right unilaterally without bounds. That was a concern on the python article.

Reply:
updated